api/subscriptions/message_cache.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2024 The VeChainThor developers
+//
+// Distributed under the GNU Lesser General Public License v3.0 software license, see the accompanying
+// file LICENSE or <https://www.gnu.org/licenses/lgpl-3.0.html>
+
+package subscriptions
+
+import (
+	"fmt"
+	"sync"
+
+	lru "github.com/hashicorp/golang-lru"
+	"github.com/vechain/thor/v2/thor"
+)
+
+type messageCache struct {
```

Comment:
Seems not necessarily creating a new type here, make it in-place within xxx_reader sounds better to me. wdyt?

Reply:
We can do that for sure, but the `newXXXReader` gets initialised for each new subscription, so we'd have to think about how we handle the cache outside of the individual subs

Reply:
You are right, I have missed that part.